p2p/net/udp.go

```diff
@@ -93,24 +88,24 @@ func (n *UDPNet) initSession(remote p2pcrypto.PublicKey) NetworkSession {
 	return session
 }
 
-// Send writes a udp packet to the target with the given data
-func (n *UDPNet) Send(to node.Node, data []byte) error {
+func NodeAddr(info *node.NodeInfo) *net.UDPAddr {
+	return &net.UDPAddr{IP: info.IP, Port: int(info.DiscoveryPort)}
+}
 
-	raddr, err := resolveUDPAddr(to.Address())
-	if err != nil {
-		return err
-	}
+// Send writes a udp packet to the target with the given data
+func (n *UDPNet) Send(to *node.NodeInfo, data []byte) error {
 
 	ns := n.cache.GetOrCreate(to.PublicKey())
 
-	if err != nil {
-		return err
-	}
-
 	sealed := ns.SealMessage(data)
 	final := p2pcrypto.PrependPubkey(sealed, n.local.PublicKey())
 
-	_, err = n.conn.WriteToUDP(final, raddr)
+	addr := NodeAddr(to)
+	if addr.IP.IsUnspecified() {
```

Comment:
add comment explaining this

Reply:
This was a mistake, removed it. thanks for that.